src/payment/strategies/barclaycard/barclaycard-payment-strategy.ts

```diff
@@ -0,0 +1,82 @@
+import { CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
+import { OrderFinalizationNotRequiredError } from '../../../order/errors';
+import { PaymentArgumentInvalidError } from '../../errors';
+import PaymentActionCreator from '../../payment-action-creator';
+import { PaymentRequestOptions } from '../../payment-request-options';
+import * as paymentStatusTypes from '../../payment-status-types';
+import PaymentStrategy from '../payment-strategy';
+
+import { BrowserInfoRequest, DeviceOS } from './barclaycard';
+
+export default class BarclaycardPaymentStrategy implements PaymentStrategy {
+    constructor(
+        private _store: CheckoutStore,
+        private _orderActionCreator: OrderActionCreator,
+        private _paymentActionCreator: PaymentActionCreator
+    ) { }
+
+    async execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment } = payload;
+
+        if (!payment) {
+            throw new PaymentArgumentInvalidError(['payment']);
+        }
+
+        const { paymentData } = payment;
+
+        await this._store.dispatch(this._orderActionCreator.submitOrder(payload, options));
+
+        return await this._store.dispatch(this._paymentActionCreator.initializeOffsitePayment(
+            payment.methodId,
+            payment.gatewayId,
+            {
+                ...paymentData,
+                formattedPayload: {
+                    ...this._getBrowserInfo(),
```

Comment:
shouldn't this be `browser_info: this._getBrowserInfo()` ? I would suggest not to merge anything that you haven't tested it actually works. Might need to wait for bigpay PR to be merged first

Reply:
By doing this the fields in are sent one by one to bigpay instead of being sent as an object, also it's easier for bigpay to grab every field  individually, and that is how it's actually implemented.

There should not be a problem for waiting for bigpay, as even if the fields are sent, they would just be ignored if there is no logic in bigpay to handle them, as bigpay is not strongly-typed as the sdk.

Reply:
yeah I was saying that because I thought bigpay is expecting a `browser_info` object https://github.com/bigcommerce/bigpay/pull/2054/files#diff-3d02eaae8a0c8141fff8fd2a0344b915R32

Reply:
Considering BigPay now _does_ have this info under the `browser_info` property, are we sure this is the correct format? Is it tested in the current state?

Reply:
The `browser_info` property on bigpay is used to send the info to the provider not to receive it so it should continue working as expected.

Reply:
If we end up keeping it like this, there's no need for the `...`:

```ts
formattedPayload: this._getBrowserInfo(),
```